### ISSUE TITLE
[Re] Predicting Dynamic Embedding Trajectory in Temporal Interaction Networks

### ISSUE THREAD

**Issue by HATON-R** [ReScience/submissions#70]
Original article:
S. Kumar, X. Zhang and J. Leskovec. Predicting Dynamic Embedding Trajectory in Temporal Interaction Networks. In: Proceedings of the 25th ACM SIGKDD international conference on Knowledge discovery and data mining. ACM. 2019.

PDF URL:
https://github.com/ComplexNetTSP/JODIE-RESCIENCE/blob/master/article.pdf
Metadata URL:
https://github.com/ComplexNetTSP/JODIE-RESCIENCE/blob/master/metadata.yaml
Code URL:
https://github.com/ComplexNetTSP/JODIE

Scientific domain: 
Machine Learning
Programming language:
Python
Suggested editor:

**Comment by rougier** [comment]
Thanks for your submission. We'll assign an editor soon.

**Comment by rougier** [comment]
@gdetor @koustuvsinha Can one of you edit this submission?

**Comment by gdetor** [comment]
@rougier I could handle this.

**Comment by rougier** [comment]
Great, thank you! I've assigned you as editor.

**Comment by gdetor** [comment]
Hi @ogrisel, would you be able to review this submission?

**Comment by gdetor** [comment]
@rougier Could I assign as reviewer someone off the reviewer's list?

**Comment by rougier** [comment]
Yes of course. If they accept and want to appear in the board, just tell me. You can also ask all reviewers at once using the `@ ReScience / reviewers` notification (without space, I don't want to broadcast here)

**Comment by gdetor** [comment]
Hi @ghost-nn-machine would you be able to review this submission?

**Comment by gdetor** [comment]
Hi @benureau Could you handle this review?

**Comment by gdetor** [comment]
Hi @koustuvsinha would you be willing to review this submission?

**Comment by gdetor** [comment]
Hi @neuronalX could you handle the review of this submission?

**Comment by neuronalX** [comment]
Hi @gdetor, thank you for the offer, but I am already too busy for the following month.

**Comment by gdetor** [comment]
Hi @damiendr Would you be available to review this submission?

**Comment by gdetor** [comment]
HI @hkashyap 
Could you handle this review? 
@ghost-nn-machine Gentle reminder

**Comment by hkashyap** [comment]
@gdetor I can review this submission.

**Comment by gdetor** [comment]
Thank you @hkashyap I'll assign you as a reviewer.

**Comment by gdetor** [comment]
Hi @hkashyap and @ghost-nn-machine Any updates?

**Comment by hkashyap** [comment]
@gdetor I will need more time, I plan to submit the review by 10/30.

**Comment by rougier** [comment]
Gentle reminder.

**Comment by gdetor** [comment]
Hi @hkashyap @ghost-nn-machine Any progress?

**Comment by HATON-R** [comment]
I trust this message finds you well. I am writing to request an update on the manuscript I submitted for review over a year ago today. Might you be able to provide me with some insight into the current status of the review process?

Perhaps we could consider the revisions as a symbolic birthday present for my article?

**Comment by rougier** [comment]
@HATON-R Very sorry for being so late in the review. I'll try to make things move forward.

**Comment by rougier** [comment]
@ReScience/reviewers Help needed for reviewing a paper machine learning/Python ! See https://github.com/ReScience/submissions/issues/70

**Comment by rougier** [comment]
@HATON-R Don't hesitate to remind us here we're late. We have not yet an automated process for tracking submission (but we'll soon have)

**Comment by HaoZeke** [comment]
I can review this as well. @rougier

**Comment by rougier** [comment]
@HaoZeke Thank you! You can start the review then. If you can do it in less than two weeks that would be wonderful.

**Comment by rougier** [comment]
@ghost-nn-machine Are you still available to do the review?

**Comment by HaoZeke** [comment]
> @HaoZeke Thank you! You can start the review then. If you can do it in less than two weeks that would be wonderful.

Sure, I'll try to get it done this weekend.

**Comment by rougier** [comment]
@hkashyap Can you update us on your review (just tell us if you can't do it such that we start looking for another reviewer)

**Comment by gdetor** [comment]
@hkashyap @HaoZeke Any progress?

**Comment by rougier** [comment]
@HaoZeke Any update?

**Comment by gdetor** [comment]
@HaoZeke @hkashyap Any updates?

**Comment by rougier** [comment]
@HATON-R @gdetor Maybe we need to find other reviewers. I can make a review and maybe @gdetor you can do the second one. Let's give us two weeks?

**Comment by gdetor** [comment]
@rougier That works for me

**Comment by gdetor** [comment]
@HATON-R @rougier Here is my review. 

Overall, the work shows that one can replicate the main results of the original. Moreover, the authors go the extra mile and show how the model's basic hyperparameters affect its performance. 

## Text
  - Although the authors refer to slightly different conclusions than those of the original article, they do not mention them in the summary.
  -  The mathematical symbol for the Haramard product is $\odot$.
  -  In the t-batch section, the authors mention that a batch cannot contain the same entity several times. Please be more specific.
  -  The two paragraphs "When we want to predict ... cross-entropy expressed in the following equation:" on page 7 can be merged. The authors repeat themselves.  
  -  One can mistake the number of classes C for the number of t-batches in algorithm 1. 
  -  In the first paragraph of Section  3.5, the authors repeat themselves. Please rephrase that paragraph. 
  -  When presenting their results, it might be better if the authors used the same format for the tables as in the original article. They can add their results in bold font and as a row in the table.
  -  The authors claim their results are marginally worse than the ones reported in the original work. However, it's not marginal because, for instance, the RRN performs better on the Reddit and Wikipedia datasets. 


##  Source Code
Unfortunately, due to Ray incompatibility, I couldn't run the code and verify the results. The authors have used an older version of Ray, so please update it or impose the exact version in the requirements.txt. 
  - Please add a LICENSE file
  - The code is not documented; please add some essential documentation, at least in the main components of the source code
  The source code should follow Python's PEP-8 protocol. The autopep-8 tool can help you make your scripts comply.
  - There are imported packages that are not used in the code (e.g., ray tune, ray.air in file train.py)
  - The code does not run on Pytorch 2.0 (the authors should fix that since many people use that version instead of the 1.10)
  - Did the authors try to use any specialized search algorithm with Ray Tune, such as Optuna, or do they rely only on Ray's default algorithms? 
  -  The requirement for Torch is broken. Use the -i link and then the name of the package.

**Comment by gdetor** [comment]
@rougier Gentle reminder

**Comment by rougier** [comment]
Sorry for such a long delay.

I've question to @HATON-R. Your write in the article _We use the publicly available code provided by the authors to reproduce their results and validate their conclusions. Our code is available at https://github.com/ComplexNetTSP/JODIE._ 
And later your wrte _We re‐implemented the JODIE model using a newer version of Pytorch [9] (version 1.10 and python 3.8)_

Does that mean you re-used the original code? Or did you start from the article description to try to write a brand new version ? Both are acceptable but in the first case, this is a reproduction and in the second case this is a replication. The difference is important for the journal.

@gdetor Don't hesitate to email me directly if I'm late. I need to put back mail notification for GitHub.

**Comment by gdetor** [comment]
@rougier Gentle reminder

**Comment by rougier** [comment]
Thanks for the reminder. @HATON-R Could you answer the questions above?

**Comment by gdetor** [comment]
@HATON-R Any updates ?

**Comment by gdetor** [comment]
Hi @HATON-R Any progress on this matter?

**Comment by rougier** [comment]
Closing until further notice from @HATON-R. Feel free to-reopen.
